tests/ui/str_to_string.rs

```diff
@@ -0,0 +1,5 @@
+#![warn(clippy::str_to_string)]
+
+fn main() {
+    let _ = "hello world".to_string();
```

Comment:
There should be a test, where the `&str` isn't a literal.

Reply:
ok